src/dataviews/category-dataview-model.js

```diff
@@ -0,0 +1,339 @@
+    },
+    DataviewModelBase.prototype.defaults
+  ),
+
+  url: function () {
+    return this.get('url') + '?bbox=' + this.get('boundingBox') + '&own_filter=' + (this.get('locked') ? 1 : 0);
+  },
+
+  initialize: function (attrs, opts) {
+    this._data = new CategoriesCollection();
+
+    DataviewModelBase.prototype.initialize.call(this, attrs, opts);
+
+    // Locked categories collection
+    this.locked = new LockedCatsCollection();
+
+    // Internal model for calculating total amount of values in the category
+    this.rangeModel = new CategoryModelRange();
+
+    // Search model
+    this.search = new SearchModel({}, {
+      locked: this.locked
+    });
+  },
+
+  // Set any needed parameter when they have changed in this model
+  _setInternalModels: function () {
+    var url = this.get('url');
+
+    this.search.set({
+      url: url,
+      boundingBox: this.get('boundingBox')
+    });
+
+    this.rangeModel.setUrl(url);
+  },
+
+  _onChangeBinds: function () {
+    this._setInternalModels();
+
+    this.rangeModel.bind('change:totalCount change:categoriesCount', function () {
+      this.set({
+        totalCount: this.rangeModel.get('totalCount'),
+        categoriesCount: this.rangeModel.get('categoriesCount')
+      });
+    }, this);
+
+    this.bind('change:url', function () {
+      if (this.get('sync') && !this.isDisabled()) {
+        this._fetch();
+      }
+    }, this);
+
+    this.bind('change:boundingBox', function () {
+      // If a search is applied and bounding bounds has changed,
+      // don't fetch new raw data
+      if (this.get('bbox') && !this.isSearchApplied() && !this.isDisabled()) {
+        this._fetch();
+      }
+    }, this);
+
+    this.bind('change:url change:boundingBox', function () {
+      this.search.set({
+        url: this.get('url'),
+        boundingBox: this.get('boundingBox')
+      });
+    }, this);
+
+    this.bind('change:disabled', function (mdl, isDisabled) {
+      if (!isDisabled) {
+        if (mdl.changedAttributes(this._previousAttrs)) {
+          this._fetch();
+        }
+      } else {
+        this._previousAttrs = {
+          url: this.get('url'),
+          boundingBox: this.get('boundingBox')
+        };
+      }
+    }, this);
+
+    this.locked.bind('change add remove', function () {
+      this.trigger('change:lockCollection', this.locked, this);
+    }, this);
+
+    this.search.bind('loading', function () {
+      this.trigger('loading', this);
+    }, this);
+    this.search.bind('sync', function () {
+      this.trigger('sync', this);
+    }, this);
+    this.search.bind('error', function (e) {
+      if (!e || (e && e.statusText !== 'abort')) {
+        this.trigger('error', this);
+      }
+    }, this);
+    this.search.bind('change:data', function () {
+      this.trigger('change:searchData', this.search, this);
+    }, this);
+  },
+
+  getLockedSize: function () {
```

Comment:
As mentioned [here](https://github.com/CartoDB/cartodb.js/pull/989/files#r49316685), all these methods below should be part of widget models IMO. From what I'm seeing, widget views in the deep-insights repo are using these, which is probably a sign that they are more widget related.